packages/web3-react/src/components/ConnectModal/index.tsx

```diff
@@ -61,23 +61,19 @@ const ConnectModal: React.FC<{
     { id: 'PROFILE', content: <Profile /> }
   ]
 
-  function hide() {
+  const hide = useCallback(() => {
     context.setOpen(false)
-  }
+  }, [context])
 
   useEffect(() => {
-    if (isConnected) {
-      if (context.route !== routes.PROFILE) {
-        hide() // Hide on connect
-      }
-    } else {
-      hide() // Hide on connect
+    if (isConnected && context.route !== routes.PROFILE) {
```

Comment:
Is it intentional to skip `hide()` when `isConnected` is false?

Reply:
Yes, because the previous dependency list only included `isConnected`, it would only execute when `isConnected` changes. What we need is to hide this modal when the wallet is connected and the route is not `profile`, and keep it unchanged in other scenarios.

Reply:
Got it, thanks for the explanation :+1: